### PR TITLE
Adding unit testing for attribute descriptions

### PR DESCRIPTION
**Pull request by arcturus2**
### Description of the Change

This is related to #778 Review default values for attribute descriptions. Unit testing has also been added for all attribute descriptions and there where a couple of cases picked up in the unit testing where the actual default values were different for some scenarios. This has also been fixed by updating the attribute descriptions.

Note that there are some methods I was not sure how to test so they have been commented. The `hashCode` methods in most attribute descriptions have also been commented because they are hard to test.

### Benefits

With the addition of the unit tests we should have a better code coverage on the attribute descriptions which are a fundamental component.

### Applicable Issues

#778

### CODE REVIEW

**Comment by arcturus2** [comment]
I've still got more to go but I wanted to see how well the code coverage is going with the default generated files.

File: CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of setDefault method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testSetDefault() {
+        Object value = null;
+        instance.setDefault(value);
```
Comment (aldebaran30701) [inline CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java:106]: Should this have an assert to check if it was set correctly?

File: CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of setCapacity method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testSetCapacity() {
+        int capacity = 0;
+        instance.setCapacity(capacity);
```
Comment (aldebaran30701) [inline CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java:125]: Should this have an assert to check if it was set correctly?

File: CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+    }
+
+    /**
+     * Test of setByte method, of class BooleanAttributeDescription.
+     */
+    @Test
+    public void testSetByte() {
+        int id = 0;
+        byte value = 0;
+        instance.setByte(id, value);
```
Comment (aldebaran30701) [inline CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java:146]: Should this have an assert to check if it was set correctly?
Quite a few more it seems, I wont add review for all. If it does need attention then I'd say just go through the tests.

**Review by aldebaran30701 — CHANGES_REQUESTED** [review]
Set methods don't seem to check if they were set correctly

**Comment by arcturus2** [comment]
> 
> 
> Set methods don't seem to check if they were set correctly

Your right, they should. I started doing this for some but I will go back and do it for all of these.

File: CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java (hunk 1 above)
Comment (arcturus2) [reply CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java:106]: Updated thanks so much for the feedback

File: CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java (hunk 2 above)
Comment (arcturus2) [reply CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java:126]: Updated

File: CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java (hunk 3 above)
Comment (arcturus2) [reply CoreGraphFramework/test/unit/src/au/gov/asd/tac/constellation/graph/attribute/BooleanAttributeDescriptionNGTest.java:148]: Yep updated most of them and ready for a review @aldebaran30701. I might continue with more tests down the track but would like to get whats done in if thats ok.

**Comment by arcturus2** [comment]
@aldebaran30701 sorry it looks like I've missed some assertions in the setters. I will try to get this sorted today.

**Comment by arcturus2** [comment]
Ready for a review please. I have made sure all setters have the asserts. There are still some further testing that can be done but happy with what we have and being able to fix a couple of bugs/inconsistencies that have arisen.

**Review by antares1470 — APPROVED** [review]

**Review by aldebaran30701 — APPROVED** [review]

**Pull request merged**